eng/pipelines/templates/steps/test-packages.yml

```diff
@@ -0,0 +1,111 @@
+parameters:
+  - name: ServiceDirectory
+    type: string
+    default: auto
+  - name: Toolchain
+    type: string
+    default: stable
+  - name: ArtifactSuffix
+    type: string
+    default: "linux"
+  - name: PublishArtifacts
+    type: boolean
+    default: false
+  - name: UnitTests
+    type: boolean
+    default: false
+  - name: FunctionalTests
+    type: boolean
+    default: false
+  - name: TestTimeoutInMinutes
+    type: number
+    default: 60
+
+steps:
+  # using -Zscript with verify-dependencies.rs requires nightly be installed
+  - ${{ if ne(parameters.Toolchain, 'nightly') }}:
+      - template: /eng/pipelines/templates/steps/use-rust.yml@self
+        parameters:
+          Toolchain: nightly
+
+  - template: /eng/pipelines/templates/steps/use-rust.yml@self
+    parameters:
+      Toolchain: ${{ parameters.Toolchain }}
+
+  - template: /eng/common/pipelines/templates/steps/set-default-branch.yml@self
+
+  - script: |
+      echo "##vso[build.addbuildtag]Scheduled"
+    displayName: "Tag scheduled builds"
+    condition: and(eq(variables['Build.SourceBranchName'], variables['DefaultBranch']), eq(variables['Build.Reason'],'Schedule'))
+
+  # now we need to call Save-Package-Properties so that we can filter on it
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
@scbedd @hallipr we might want to consider adding this logic to eng/common.

Reply:
Agreed.   It's a big chunk of reusable yaml

Reply:
I've started a draft pr:
https://github.com/Azure/azure-sdk-tools/pull/8900